src/Common/FileSegment.cpp

```diff
@@ -646,6 +646,13 @@ void FileSegment::assertNotDetached() const
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Operation not allowed, file segment is detached");
 }
 
+void FileSegment::assertDetachedStatus() const
+{
+    assert(
+        (download_state == State::EMPTY) || (download_state == State::PARTIALLY_DOWNLOADED)
```

Comment:
`State::PARTIALLY_DOWNLOADED` should be redundant in this place.
Let's update the file segment state in completeImpl right before this line https://github.com/ClickHouse/ClickHouse/blob/5801de7f723bf06ca3156554a5ea2105b054bf40/src/Common/FileSegment.cpp#L565
to `State::PARTIALLY_DOWNLOADED_NO_CONTINUATION`.